base/cvd/cuttlefish/host/libs/config/fetched_archive.cc

```diff
@@ -44,8 +46,8 @@ Result<FetchedArchive> FetchedArchive::Create(
     const FetcherConfig& fetcher_config, FileSource source,
     std::string_view archive) {
   std::optional<ReadableZip> zip_file;
-  std::set<std::string> members;
-  std::map<std::string, std::string> extracted_members;
+  std::set<std::string, std::less<void>> members;
```

Comment:
Why explicitly provide the comparator here and below? Won't it end up calling `operator<` on `std::string` anyways, being the same as `std::set<std::string>`?

Reply:
Using the [`std::less<void>`](https://en.cppreference.com/w/cpp/utility/functional/less_void.html) comparator allows `FetchedArchive::MemberFilepath` to use `members_.count` and `extracted_.find` with `std::string_view` arguments, even though the container holds a `std::string`. The default comparator of a `std::set<K>` or a `std::map<K, ...>` is `std::less<K>`, which has a parameter type of `const K&`. In this case `K` is `std::string`, which could force an allocation the caller only has a `std::string_view`.